cmd/podman/containers/createspec.go

```diff
@@ -0,0 +1,98 @@
+package containers
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/containers/common/pkg/completion"
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/containers/podman/v4/pkg/specgen"
+	"github.com/containers/podman/v4/pkg/specgenutil"
+	"github.com/spf13/cobra"
+)
+
+var (
+	createSpecDescription = `Creates a new container from the given json file cntaining a filled out specgen`
+	createSpecCommand     = &cobra.Command{
+		Use:               "createspec [options] FILE",
+		Short:             "Create a new container from json",
+		Long:              createSpecDescription,
+		RunE:              createspec,
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: completion.AutocompleteNone,
+		Example:           `podman container createspec --cpus=5 ~/Documents/spec.json`,
+	}
+)
+
+var (
+	containerOpts entities.ContainerCreateOptions
+	startCtr      bool
+)
+
+func createSpecFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+
+	startFlagName := "start"
+	flags.BoolVar(&startCtr, startFlagName, false, "start the new container")
+
+	flags.SetInterspersed(false)
+	common.DefineCreateDefaults(&containerOpts)
+	common.DefineCreateFlags(cmd, &containerOpts, false, false)
+	common.DefineNetFlags(cmd)
+}
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: createSpecCommand,
+		Parent:  containerCmd,
+	})
+
+	createSpecFlags(createSpecCommand)
+}
+
+func createspec(cmd *cobra.Command, args []string) error {
+	f, err := os.Open(args[0])
+	if err != nil {
+		return err
+	}
+
+	data, err := os.ReadFile(f.Name())
+	if err != nil {
+		return err
+	}
+
+	spec := &specgen.SpecGenerator{}
+
+	err = json.Unmarshal(data, spec)
+	if err != nil {
+		return err
+	}
+
+	err = specgenutil.FillOutSpecGen(spec, &containerOpts, []string{})
```

Comment:
Is there any chance that spec will get overridden by containerOpts?

Reply:
no, I have adjusted `FillOutSpecGen` in the last few months to only put a value in the spec if the user has explicitly specified || the spec item is empty.